src/main/java/movies/Server.java

```diff
@@ -109,23 +111,26 @@ private static Map<CrewRole, Long> crewCountForMovie(List<Credit> credits) {
 	}
 
 	private static Object moviesEndpoint(Request req, Response res) {
-		var movies = MOVIES.get().stream();
+		var movies = MOVIES.get();
 		movies = sortByDescReleaseDate(movies);
 		var query = req.queryParamOrDefault("q", req.queryParams("query"));
 		if (query != null) {
-			movies = movies.filter(m -> Pattern.matches(".*" + query.toUpperCase() + ".*", m.title.toUpperCase()));
+			movies = movies.stream().filter(m -> Pattern.matches(".*" + query.toUpperCase() + ".*", m.title.toUpperCase())).toList();
 		}
 		return replyJSON(res, movies);
 	}
 
-	private static Stream<Movie> sortByDescReleaseDate(Stream<Movie> movies) {
-		return movies.sorted(Comparator.comparing((Movie m) -> {
+	private static List<Movie> sortByDescReleaseDate(List<Movie> movies) {
+		var sortedMovies = new ArrayList<Movie>(movies);
+		Collections.sort(sortedMovies, Comparator.comparing((Movie m) -> {
```

Comment:
Minor: I suspect this can be `sortedMovies.sort(Comparator)` instead of `Collections.sort`.

(Not entirely sure if there's any performance difference or not -- I'm guessing it's equivalent)

Reply:
thanks - I think it's cleaner so changing it :)